modules/renter/hostdb/scan.go

```diff
@@ -153,11 +159,9 @@ func (hdb *HostDB) threadedProbeHosts() {
 			hostEntry.reliability = MaxReliability
 			hostEntry.weight = calculateHostWeight(*hostEntry)
 
-			// If the host is not already in the database and 'MaxActiveHosts' has not
-			// been reached, add the host to the database.
-			_, exists1 := hdb.activeHosts[hostEntry.NetAddress]
-			_, exists2 := hdb.allHosts[hostEntry.NetAddress]
-			if !exists1 && exists2 && len(hdb.activeHosts) < MaxActiveHosts {
+			// If 'MaxActiveHosts' has not been reached, add the host to the
+			// activeHosts tree.
+			if _, exists := hdb.activeHosts[hostEntry.NetAddress]; !exists && len(hdb.activeHosts) < MaxActiveHosts {
 				hdb.insertNode(hostEntry)
 			}
 		}
```

Comment:
this locking scope should probably be a function, it's pretty large and has conditions where the lock needs to be released inside the scope. I think a safer construction would be the 

```
func() {
    lock
    defer unlock
}()
```

pattern. there's a lot of code here.